basic/src/basic.cpp

```diff
@@ -26,9 +26,9 @@ Basic& Basic::getInstance()
 
 double Basic::particleRestMass(const int particleID) const
 {
-	if (particleID < 0 || size_t(particleID) >= m_particleMass.size())
+	if (particleID < 1 || size_t(particleID) >= m_particleMass.size())
```

Comment:
Change upper border check from >= to >

Reply:
PAMA is 1-indexed, so particleId must be >0 and <= size() in order for particleId-1 to be in range of the 0-indexed array that C++ sees.